Achievements Page UI redesign

ssha0054 opened the pull request:
Co-Authored-By: @andrwcao  

# Description 
Changed the styling of the Achievements Page to make it look more consistent with the colours used in the Landing Page. Main changes are the buttons which now use styled-components and also some colour changes on the cards. The colours used here are not final and can easily be changed once the other pages are restyled and we can ensure everything is consistent.


# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 
- [x] Improvement


# Problem
The Achievements page was previously quite plain and inconsistent with how the Landing Page looks.


# Solution description
Used styled-components to create better looking buttons and used them to replace the buttons on this page. (Note: We saw  PR #231 and decided to create the same styled-components file and add onto it with some new button designs we used on this page.) Added some css to make the cards and forms look more consistent.

# Tests 
All unit tests pass? Yes

Builds successfully? Yes


Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
![image](https://user-images.githubusercontent.com/80957318/134145917-65dfc0a4-a8fb-4ec5-b5ab-2a8ced187ddb.png)

When you click on the Add Achievement button you get this:
![image](https://user-images.githubusercontent.com/80957318/134145999-f206b844-561d-43db-a211-adf3b3526f29.png)

When you click on the options button (those three dots) on the side of an achievement:
![image](https://user-images.githubusercontent.com/80957318/134146074-7cb5ca75-abad-44a1-aaab-708a20342ea3.png)

When you click on the edit button:
![image](https://user-images.githubusercontent.com/80957318/134146125-babdda94-a0bf-452c-948e-bd7b6d5306be.png)

When you click on the delete button:
![image](https://user-images.githubusercontent.com/80957318/134146164-d06b1a36-8ff5-4b16-9b1d-d74b1c5a9203.png)


# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low - Only changed UI for one page. Messed around with functionality and it wasn't affected.

pmyip3 commented on client/src/components/shared/styledComponents.js:
```diff
@@ -0,0 +1,111 @@
+    padding: .375rem .75rem;
+    border: 1px solid #56658a;
+    border-radius: .25rem;
+    background-color: #AB9671;
```
same here, may be better to use red for delete button

andrwcao replied:
After merging your UI changes from main, I reckon it looks a lot better if the option's delete button is gold as it matches with the sidebar (rather than red which is pre jarring). What does everyone else think?
![image](https://user-images.githubusercontent.com/80874259/134290715-c2b3cb42-2f0e-4af7-8072-6e3ec2afe8dc.png)

ssha0054 replied:
Yeah tbh I prefer this gold colour for that delete icon over having it red as well.

ssha0054 replied:
Changed the delete button to red for consistency with the other pages but will still keep the style for the gold button commented out incase we wish to use it later.

pmyip3 commented on package.json:
```diff
@@ -5,8 +5,8 @@
-    "start": "concurrently \"cd api && node src/server.js\" \"cd client && npm start\"",
-    "dev": "concurrently \"cd api && node src/server.js\" \"cd client && npm start\""
+    "start": "concurrently \"cd api && node src/index.js\" \"cd client && npm start\"",
+    "dev": "concurrently \"cd api && node src/index.js\" \"cd client && npm start\""
```
Is there any reason we are changing it?

jfan0010 replied:
It's the change from [PR#233](https://github.com/Researchify/Researchify/pull/233) to fix the `npm run start` in researchify root folder. I believe changes in this file will not be shown after that PR is merged.

ssha0054 replied:
Yeah I just changed it so I could run the application using the single command but PR #233 has now been merged so this change here isn't regarding anything from this PR.

andrwcao commented on client/src/components/achievements/Achievement.js:
```diff
@@ -101,23 +106,27 @@ const Achievement = ({ achievement }) => {
+          </div>
+          <div>
+            <PrimaryButton
+              onClick={handleDelete}
```
Done.

rpra0006 reviewed (approved):
1. Could you maybe move the "Add Achievement" button on the left side (beneath title)? I just feel like for consistency team members and publications page have their "add" buttons on the left.

2. Not quite sure if we're going with red or gold for the delete button. In team members page we have red, but in publications we have reddish-brown. I personally prefer the red over the gold for delete, but maybe other people can have input on this?

Just some styling comments, else functionality wise and design looks good!

ssha0054 commented:
Update: Moved the add achievements button to be beneath the title and also changed the delete button to be red. Not sure what the final colour will be like but I guess making it red now will make it consistent with the other pages for now. Have kept the style for the gold button commented out incase we want to use that later. Achievements page now looks like this:
![image](https://user-images.githubusercontent.com/80957318/134336980-0132ab2a-83f0-464f-a970-0a03c07f6e71.png)

aksh0001 commented on client/src/components/shared/styledComponents.js:
```diff
@@ -81,3 +81,44 @@ export const DangerButton = styled.button`
+  }
+`;
+
+// export const OptionDeleteButton = styled.button` // Gold with darker gold background when hovering
+//     padding: .375rem .75rem;
+//     border: 1px solid #56658a;
+//     border-radius: .25rem;
+//     background-color: #AB9671;
+//     color: white;
+//     transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
+//     font-size: ${(props) => props.fontSize}; //.875rem;
+//   &:hover{
+//     background-color: #8c7a5b;
+//     color: white;
+//   }
+// `;
```
remove if unused

aksh0001 commented:
> Update: Moved the add achievements button to be beneath the title and also changed the delete button to be red. Not sure what the final colour will be like but I guess making it red now will make it consistent with the other pages for now. Have kept the style for the gold button commented out incase we want to use that later. Achievements page now looks like this:
> ![image](https://user-images.githubusercontent.com/80957318/134336980-0132ab2a-83f0-464f-a970-0a03c07f6e71.png)

Actually, just saw this now. The alignment looks a little off to me and it doesn't seem too discoverable/pleasing. What if we got rid of the `Achievements` title on the page entirely? Seems a bit redundant and I noticed the publications page doesn't have the title anymore also.

Thoughts by others?

ssha0054 commented:
> > Update: Moved the add achievements button to be beneath the title and also changed the delete button to be red. Not sure what the final colour will be like but I guess making it red now will make it consistent with the other pages for now. Have kept the style for the gold button commented out incase we want to use that later. Achievements page now looks like this:
> > ![image](https://user-images.githubusercontent.com/80957318/134336980-0132ab2a-83f0-464f-a970-0a03c07f6e71.png)
> 
> Actually, just saw this now. The alignment looks a little off to me and it doesn't seem too discoverable/pleasing. What if we got rid of the `Achievements` title on the page entirely? Seems a bit redundant and I noticed the publications page doesn't have the title anymore also.
> 
> Thoughts by others?

Yeah not sure if removing the title will be better, but the current alignment here is now consistent with what is on the team members page. Was hoping to get this PR (which was a high priority one and also PR #242 if possible) merged in before the UAT, which I believe is happening tomorrow, and then from the comments we receive from this along with feedback from more people within our project we can make changes to this in a separate PR. Don’t want to miss out on receiving feedback of this page from the UAT so will merge it in now.
